### PR TITLE
[feat] Implement target branch functionality

### LINKED ISSUE #23: Implement target branch functionality
With the `sanitize-repo` command, it should be possible to specify a `--target-branch`. This will sanitize the current branch (without committing on the current branch), and then make the commit on the target branch.

### PR DESCRIPTION
**Pull request by slarse**
Fix #23 

This PR adds the `--target-branch branch` option that sanitizes the files on the current branch, and then commits the sanitized files (along with all other files in the repo) to the target branch.

This is mutually exclusive to the new option `--no-commit`, which is functionally equivalent to the old functionality of just sanitizing the files in the current directory without creating a commit. 

I also set the `user.name` and `user.email` attributes in Git to allow for commits to be made.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[[source]]
```
Comment (tohanss) [inline Pipfile:1]: Pipfile should not be in the commit

File: .vim/coc-settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "python.formatting.provider": "black"
```
Comment (tohanss) [inline .vim/coc-settings.json:2]: This file should not be in the commit

Comment (slarse) [reply .vim/coc-settings.json:2]: Indeed

File: repobee_sanitizer/_sanitize_repo.py [hunk 3]
```diff
@@ -51,23 +57,98 @@ def create_extension_command(self) -> plug.ExtensionCommand:
...
-            metavar="<path>",
+            metavar="path",
             required=True,
         )
         parser.add_argument(
+            "-r",
             "--repo-root",
             help="Path to the worktree root of the repository to sanitize.",
             type=pathlib.Path,
+            metavar="path",
```
[the remark this answers is not in the document]
Comment (slarse) [reply repobee_sanitizer/_sanitize_repo.py:73]: When you run `--help`, metavar is the variable in the help section that signifies this argument. For example, the help section may look like this:

```
-r path --repo-root path     Path to the worktree root of the repository to sanitize.
```
If metavar is instead YAY, then it looks like this:
```
-r YAY --repo-root YAY     Path to the worktree root of the repository to sanitize.
```

If we don't metavar, then it just takes the long argument:

```
-r REPO_ROOT --repo-root REPO_ROOT     Path to the worktree root of the repository to sanitize.
```

[the remark this answers is not in the document]
Comment (slarse) [reply repobee_sanitizer/_sanitize_repo.py:73]: I've not decided if `path` or `<path>` is better

File: repobee_sanitizer/_sanitize_repo.py [hunk 4]
```diff
@@ -51,23 +57,98 @@ def create_extension_command(self) -> plug.ExtensionCommand:
...
             parser=parser,
             name="sanitize-repo",
             help="Sanitize the current repository.",
             description="Sanitize the current repository.",
             callback=self._sanitize_repo,
         )
+
+
+def _sanitize_files(
+    root: pathlib.Path, file_relpaths: List[pathlib.Path]
```
[the remark this answers is not in the document]
Comment (slarse) [reply repobee_sanitizer/_sanitize_repo.py:100]: Possibly, rename to `basedir`

File: tests/test_ext_commands.py [hunk 5]
```diff
@@ -129,3 +121,11 @@ def fake_repo(tmpdir) -> _FakeRepoInfo:
         file_infos=file_infos,
         default_branch=default_branch,
     )
+
+
+def assert_expected_text_in_files(file_infos):
```
[the remark this answers is not in the document]
Comment (slarse) [reply tests/test_ext_commands.py:175]: Internal test function, you will always have the source available. Type hinting this is a little bit more effort than its worth, imo.

**Pull request merged**